relay-server/src/services/processor/span.rs

```diff
@@ -31,21 +31,12 @@ pub fn filter(state: &mut ProcessEnvelopeState<SpanGroup>) {
 }
 
 /// Creates a span from the transaction and applies tag extraction on it.
-pub fn extract_transaction_span(event: &Event) -> Span {
-    let mut transaction_span: Span = event.into();
+///
+/// Returns `None` when [`tag_extraction::extract_span_tags`] clears the span, which it shouldn't.
+pub fn extract_transaction_span(event: &Event, max_tag_value_size: usize) -> Option<Span> {
```

Comment:
Not too happy with the `Option` here, but there is no guarantee that `extract_span_tags` does not set the `Annotated<Span>` to empty.